src/components/Search/Search.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {TeamsType, UserDataType} from 'types/types';
+import {SearchContainer, SearchInput} from './Search.styled';
+
+interface SearchProps<ObjectType> {
+    originalObject: ObjectType[];
+    updateFilteredObject: (updatedObject: ObjectType[]) => void;
+    notifyError: (hasError: boolean) => void;
+    searchProps: (keyof ObjectType)[];
+    placeholder: string;
+}
+
+const Search = <ObjectType extends TeamsType | UserDataType>(props: SearchProps<ObjectType>): JSX.Element => {
+    const {originalObject, updateFilteredObject, notifyError, searchProps, placeholder} = props;
+
+    const [searchValue, setSearchValue] = React.useState<string>('');
+
+    const handleSearch = React.useCallback(
```

Comment:
Re-written the component using the hooks `useCallback` and `useMemo`, instead of `useEffect`, this prevents all rerendering issues that the component was getting with the previous approach, while not getting the exhausted deps issue with eslint, and allows for better performance in the search component